google_guest_agent/main.go

```diff
@@ -119,6 +119,28 @@ func runUpdate(ctx context.Context) {
 	wg.Wait()
 }
 
+func recordTelemetry(ctx context.Context, mdsClient *metadata.Client) {
+	for {
+		if newMetadata != nil && !newMetadata.Instance.Attributes.DisableTelemetry && !newMetadata.Project.Attributes.DisableTelemetry {
+			d := telemetry.Data{
+				AgentName:     programName,
+				AgentVersion:  version,
+				AgentArch:     runtime.GOARCH,
+				OS:            runtime.GOOS,
+				LongName:      osInfo.PrettyName,
+				ShortName:     osInfo.OS,
+				Version:       osInfo.VersionID,
+				KernelRelease: osInfo.KernelRelease,
+				KernelVersion: osInfo.KernelVersion,
+			}
+			if err := telemetry.Record(ctx, mdsClient, d); err != nil {
+				logger.Debugf("Error recording telemetry: %v", err)
+			}
+		}
+		time.Sleep(24 * time.Hour)
```

Comment:
Nope, please, please use the [just introduced scheduler](https://github.com/GoogleCloudPlatform/guest-agent/tree/main/utils/scheduler)

Reply:
Hi Leo, your link looks like a 404 not found

Reply:
Please use this [updated link](https://github.com/GoogleCloudPlatform/guest-agent/tree/main/google_guest_agent/scheduler)

Reply:
I can update this to use that but I'm a little concerned with that library in general, calling Get has unexpected side effects, and you are importing a library I don't think you really need. 
Writing scheduling logic should be pretty straightforward, especially since I don't think you need any of the complexities of that library? Also you probably should have your main configure your scheduler and not have the idea of default jobs, that can easily cause issues for you.